src/processor/graph-update-queue-consumer.service.ts

```diff
@@ -38,20 +38,48 @@ export class QueueConsumerService extends WorkerHost implements OnApplicationBoo
   }
 
   constructor(
-    @InjectQueue('graphUpdateQueue') private graphUpdateQueue: Queue,
-    private cacheManager: ReconnectionCacheMgrService,
-    private graphSdkService: ReconnectionGraphService,
-    private blockchainService: BlockchainService,
-    private configService: ConfigService,
-    private schedulerRegistry: SchedulerRegistry,
-    private eventEmitter: EventEmitter2,
+    @InjectQueue(ReconnectionServiceConstants.GRAPH_UPDATE_QUEUE_NAME) private readonly graphUpdateQueue: Queue,
+    @InjectQueue(ReconnectionServiceConstants.TX_MONITOR_QUEUE_NAME) private readonly txMonitorQueue: Queue,
+    private readonly cacheManager: ReconnectionCacheMgrService,
+    private readonly graphSdkService: ReconnectionGraphService,
+    private readonly blockchainService: BlockchainService,
+    private readonly configService: ConfigService,
+    private readonly schedulerRegistry: SchedulerRegistry,
   ) {
     super();
     this.logger = new Logger(this.constructor.name);
     graphUpdateQueue.pause(); // pause queue until we're ready to start processing
   }
 
-  async process(job: Job<IGraphUpdateJob, any, string>) {
+  async process(job: Job<any, any, string>): Promise<void> {
+    switch (job.data.state) {
+      case GraphUpdateJobState.Unprocessed:
+        return this.updateGraph(job);
+
+      case GraphUpdateJobState.Submitted:
+        this.logger.error(`Processing job ${job.id} in state 'Submitted'; should not happen`);
+        throw new UnrecoverableError('Processing job in Submitted state');
+        // eslint-disable-next-line no-unreachable
+        break;
+
+      case GraphUpdateJobState.ChainFailureNoRetry:
+        await job.moveToFailed(new Error('Fatal job error condition after chain submission'), job.token!);
+        break;
+
+      case GraphUpdateJobState.ChainFailureRetry:
+        await job.updateData({ ...job.data, state: GraphUpdateJobState.Unprocessed });
+        throw new Error('Error detected on-chain; will retry if allowed');
+        // eslint-disable-next-line no-unreachable
+        break;
+
+      case GraphUpdateJobState.MonitorSuccess:
+      default:
+    }
+
+    return Promise.resolve();
```

Comment:
nice, i like all the beautiful updates , thanks

Reply:
🙏🏻